src/methods/properties.js

```diff
@@ -117,15 +117,15 @@ function InternalUpdatedProperty(realm: Realm, O: ObjectValue, P: PropertyKeyVal
   let propertyBinding = InternalGetPropertiesMap(O, P).get(P);
   let desc = propertyBinding === undefined ? undefined : propertyBinding.descriptor;
   if (desc === undefined) {
-    if (O === realm.$GlobalObject) {
+    if (O === realm.$GlobalObject && !realm.isStrict) {
```

Comment:
Per our discussion offline, this should check both the global realm.isStrict and current FunctionValue's isStrict.